### PR TITLE
Changes in MainActivity.java to avoid using depricated code

### PR DESCRIPTION
**Pull request by pulakdp**
Code should be updated according to https://stackoverflow.com/questions/41416450/setproviders-method-is-marked-deprecated/41616963#41616963

### CODE REVIEW

**Comment by pulak03** [comment]
Code should be changed according to https://stackoverflow.com/questions/41416450/setproviders-method-is-marked-deprecated/41616963#41616963

File: app/src/main/java/com/google/firebase/udacity/friendlychat/MainActivity.java [hunk 1]
```diff
@@ -59,6 +66,8 @@
     private FirebaseDatabase mFirebaseDatabase;
     private DatabaseReference mMessagesDatabaseReference;
     private ChildEventListener mChildEventListener;
+    private FirebaseAuth mFirebaseAuth;
+    private FirebaseAuth.AuthStateListener mAuthStateListener;
```
Comment (pulak03) [inline app/src/main/java/com/google/firebase/udacity/friendlychat/MainActivity.java]: List<AuthUI.IdpConfig> providers;

**Comment by omerasif57** [comment]
as of revision 3.2.1, above samples are deprecated.
[FirebaseUI for Auth - Github](https://github.com/firebase/FirebaseUI-Android/tree/master/auth)
Fix:
```
List<AuthUI.IdpConfig> providers = Arrays.asList(
                            new AuthUI.IdpConfig.EmailBuilder().build(),
                            new AuthUI.IdpConfig.PhoneBuilder().build(),
                            new AuthUI.IdpConfig.GoogleBuilder().build(),
                            new AuthUI.IdpConfig.FacebookBuilder().build(),
                            new AuthUI.IdpConfig.TwitterBuilder().build()))
startActivityForResult(
                            AuthUI.getInstance()
                                .createSignInIntentBuilder()
                                .setAvailableProviders(providers)
                                .build(),
                            RC_SIGN_IN
                    );
```

**Comment by a-lloma** [comment]
i had this same problem
`compile 'com.firebaseui:firebase-ui-auth:3.2.2'`

Change the following:
`compile 'com.android.support:design:24.2.0'`
`compile 'com.android.support:appcompat-v7:24.2.0'`
to
`compile 'com.android.support:design:27.0.2'`
`compile 'com.android.support:appcompat-v7:27.0.2'`

font: [https://stackoverflow.com/questions/49178668/gradle-build-errors-with-firebase-ui-auth](url)

**Pull request closed without merging**